src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -402,9 +403,22 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
                 ScriptHost.IsFunctionDataCacheEnabled = false;
             }
 
+            if (_environment.IsApplicationInsightsAgentEnabled() ||
+                (bool.TryParse(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.WorkerApplicationInsightsLoggingEnabled), out bool appInsightsWorkerEnabled) &&
+                appInsightsWorkerEnabled))
```

Comment:
For other places, we don't really check the capability value is truthy.(ex: line 398). Should we stick to that approach for being consistent?

Reply:
I really dislike that we only check for existence of the key in a lot of places -- it means if you explicitly set something to "false", it'll still be evaluated as "true". I find that really confusing.

I didn't want to touch all that, though -- but I do think we should address it at some point. I just didn't want to add to the confusion.